src/helper/reddit_api.js

```
@@ -0,0 +1,28 @@
+import axios from 'axios';
+import moment from 'moment';
+
+// API Reference - https://reddit-api.readthedocs.io/en/latest/#searching-submissions
+
+const BASE_URL = 'https://api.pushshift.io/reddit/submission/search/';
+const NOW = moment();
+
+const getPosts = async (subreddit) => {
+  const ONE_YEAR_BEFORE = NOW.subtract(1, 'years').unix();
+  const QTY = 500;
+
+  const NEW_URL = `${BASE_URL}?subreddit=${subreddit}&after=${ONE_YEAR_BEFORE}&size=${QTY}}&sort=desc`;
+
+  try {
+    const response = await axios.get(NEW_URL);
+    if (response.status === 200) {
+      const posts = response.data.data;
+      return posts;
+    }
+    throw new Error('No such subreddit!');
```

It's this the only possible error? If yes, then it's fine.

---

if it is not, how I can display error base on responded status?

---

This actually doesn't work currently. Check the response for a non-existing subreddit. It's

```
{
    "data": []
}
```

You can try it in your app. When you enter a random string the spinner just stays there.

Since there's no real API documentation for pushshift.io there are no documented error cases as well. So I'd throw the error on when you get the response above